### ISSUE TITLE
ci: tighten Bandit SAST threshold to fail on medium+ severity issues

### ISSUE THREAD

**Issue by the404packet** [Checkora/Checkora#823]
## Problem

The current Bandit SAST step in our CI pipeline only fails on **High** severity issues. This threshold is too permissive given the current state of the project.

Additionally, there is a which I think, a latent bug in the existing grep condition, it matches `"Severity: High"` but Bandit may output `"Severity: HIGH"` (all caps) depending on the version, meaning High severity issues could silently pass through without ever blocking CI. The check I thought was protecting us may not have been working correctly at all.

## Why This Needs to Change

The project is growing in scope and sensitivity:

- **Login and registration pages** are continuously being modified and handle user credentials directly
- **User input flows** are active across multiple pages and are a common attack surface
- **Real-time collaboration** is on the roadmap, which will significantly expand the attack surface

Medium severity findings on authentication code or input handling are just as dangerous in practice as High ones. Waiting until something is flagged High before blocking CI is leaving real risk on the table, especially as new features land.

## What Needs to Be Changed

In `.github/workflows/ci.yml`, under the `sast` job, the Bandit step needs three changes:

**1. Fix the latent case-sensitivity bug:**
```bash
# Current (broken for all-caps output)
grep -q "Severity: High" bandit-report.txt

# Fix
grep -qi "severity: high" bandit-report.txt
```

**2. Extend the threshold to also catch Medium severity:**
```bash
# Current
grep -qi "severity: high" bandit-report.txt

# Change to
grep -qi "severity: high\|severity: medium" bandit-report.txt
```

**3. Update the step name to reflect the new threshold:**
```yaml
# Current
- name: Run Bandit scan (fail on HIGH only)

# Change to
- name: Run Bandit scan (fail on HIGH and MEDIUM)
```

## Impact

- CI will block merges when Bandit reports any Medium or High severity finding
- Low severity issues continue to pass without blocking
- The case-insensitive fix ensures High issues are actually caught which they may not   have been before

## Notes
This is a CI configuration-only change. No application code is affected.

**Comment by the404packet** [comment]
@itsmeesnehaa @EDWARD-012 @triemerge pls assign me this issue
